lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -262,18 +262,20 @@ SetOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
                         OpaqueProperties properties, RegionRange regions,
                         llvm::SmallVectorImpl<Type> &inferredReturnTypes) {
 
-  Value leftInput = operands[0];
-  Value rightInput = operands[1];
-
-  TypeRange leftFieldTypes = cast<TupleType>(leftInput.getType()).getTypes();
-  TypeRange rightFieldTypes = cast<TupleType>(rightInput.getType()).getTypes();
+  ValueRange inputs = operands;
 
-  if (leftFieldTypes != rightFieldTypes)
-    return ::emitError(loc.value())
-           << "left and right inputs must have the same field types";
+  if (inputs.size() < 2)
+    return ::emitError(loc.value()) << "expected at least 2 inputs";
 
-  auto resultType = TupleType::get(context, leftFieldTypes);
+  TypeRange fieldType = cast<TupleType>(inputs[0].getType()).getTypes();
```

Comment:
Nit: `fieldTypes` (to reflect that it's a range).

Reply:
Actually, maybe even `firstInputFieldTypes` or `referenceFieldTypes`, `refFieldTypes`, `expectedFieldTypes`?